src/pudl/metadata/fields.py

```diff
@@ -4257,6 +4338,10 @@
             "pattern": r"^\d{2}$",
         },
     },
+    "state_of_incorporation": {
+        "type": "string",
+        "description": "Two letter state code where company is incorporated.",
+    },
```

Comment:
Should this / can this be constrained to match only valid state abbreviations?

Reply:
I think it should and maybe can. Is there a place where we're applying such a constraint right now? The base `state` field doesn't have any constraints on it.

Reply:
Ah, you're right this is all over the place and we have yet to try and enforce it because it will probably mean chasing down issues.